src/modules/functionality.js

```diff
@@ -0,0 +1,77 @@
+const taskList = document.getElementById('taskList');
+export const addTask = (taskInput, taskList) => {
+  const task = taskInput.value;
+  const taskId = Date.now();
+  const newTask = document.createElement('div');
+  newTask.textContent = task;
+
+  const taskObject = {
+    id: taskId,
+    description: task,
+    completed: false,
+    index: taskList.childElementCount + 1,
+  };
+
+  // Create taskBox element
+  const taskBox = document.createElement('div');
+  taskBox.classList = 'task-box-css';
+  taskBox.dataset.id = taskId;
+  taskBox.innerHTML = ` <div class="task-activity">
+                    <input type="checkbox" class="input-check">
+                    <input type="text" class="text-input" value="${task}">
+                  </div>
+                  <i class="trash-icon" id="delete-icon">&#x1F5D1;</i>
+                  <i class="move-icon" id="order-icon">&#x22EE;</i>`;
+
+  // Append taskBox to taskList
+  taskList.appendChild(taskBox);
+
+  taskInput.value = '';
+
+  const tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+  tasks.push(taskObject); // push taskObject to task array
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
+
+export const updateTask = (oldTask, newTask) => {
+  const tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+  const index = tasks.findIndex((task) => task.description === oldTask);
+  if (index !== -1) {
+    tasks[index].description = newTask;
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+  }
+};
+
+export const loadTasks = () => {
+  const tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+  tasks.forEach((task) => {
+    const taskBox = document.createElement('div');
+    taskBox.classList = 'task-box-css';
+    taskBox.innerHTML = ` <div class="task-activity">
+                    <input type="checkbox" class="input-check">
+                    <input type="text" class="text-input" value="${task.description}">
+                  </div>
+                  <i class="trash-icon" id="delete-icon">&#x1F5D1;</i>
+                  <i class="move-icon" id="order-icon">&#x22EE;</i>`;
+
+    // Append taskBox to taskList
+    taskList.appendChild(taskBox);
+
+    // Add event listener for editing tasks
+    const textInput = taskBox.querySelector('.text-input');
+    textInput.addEventListener('change', (e) => {
+      const editedTask = e.target.value;
+      updateTask(task.description, editedTask);
+    });
+  });
+};
+// delete//
+export const deleteTask = (index) => {
+  const tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+  tasks.splice(index, 1);
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+  const taskBoxes = document.getElementsByClassName('task-box-css');
+  for (let i = 0; i < taskBoxes.length; i += 1) {
+    taskBoxes[i].dataset.index = i;
+  }
+};
```

Comment:
- [ ] You have implemented a lot of interesting functionalities :+1: except that this deletion is not working as expected now. There is a bug caused by indexes (indices) that are not unique in order and value as the requirement demanded (_where indexes are updated in the correct order_). This is forcing some deleted tasks to be restored when the page is refreshed because the local storage doesn't automatically sync with the client page.
- [ ] Kindly watched the attached [video](https://user-images.githubusercontent.com/99715640/230662988-5c4c8695-c1d3-402f-a7b8-8f2bf1964934.mp4) of my review where I recorded a demo of my discovery, and let's make appropriate changes to ensure optimum user experience with our app, which undoubtedly will be helpful to many users. :hammer_and_wrench:

Reply:
Thanks Abayomi, it's a big bug...I've been trying to solve it but it's very difficult. I'll keep working on it.

Reply:
@Maedea9 , sorry about that. Please do let me know via slack if after 6 hours from now, it's still not fixed so that we can look into it together. I should be available by then. Kindly let me know if it's the last for this week too 🙏

Reply:
Thank you so much @AbayomiOlaoye I fixed it! It was indeed a problem with the index that was established in my taskObject object. Now it's working :) Have a good weekend!